File: Python/gui/MainWindow.py
#!/usr/bin/env python3

from PyQt4 import QtGui
from PyQt4 import QtCore

from .PuzzleSolver import PuzzleSolver

class MainWindow ( QtGui.QMainWindow ):
   """Main window of the program"""
   def __init__ ( self ):
      super ( MainWindow , self ).__init__ ()
      # Create the main widget
      self.__puzzle_solver = PuzzleSolver ()
      # Set components
      self.setCentralWidget ( self.__puzzle_solver )
      self.setWindowTitle ( "Puzzle solver for Python v3" )

File: CPP/src/gui/mainwindow/mainwindow.cc
# include "mainwindow.hh"

Gui::MainWindow::MainWindow ( void )
   : QMainWindow ()
{
   buildGui ();
}

Gui::MainWindow::~MainWindow ( void )
{
}

void Gui::MainWindow::buildGui ( void )
{
   puzzlesolver = new Gui::PuzzleSolver ();
   
   this->setCentralWidget ( puzzlesolver );
   
   return;
}
